modules/task/execution_test.go

```diff
@@ -114,21 +117,33 @@ func TestEnqueueWithoutRegister(t *testing.T) {
 }
 
 func TestConsumeWithoutRegister(t *testing.T) {
+	host, _ := service.NewScopedHost(service.NopHost(), "task", "hello")
+	_testScope = host.Metrics()
+	_globalBackend = NewInMemBackend(host)
+	_errorCh = _globalBackend.(*inMemBackend).ErrorCh()
```

Comment:
It does not really matter how different backends implement propagating the error. In both cases we are checking the error manifests as expected irrespective of whether is comes through a channel or a log message. In memory backend is primarily used for testing and channels make it easy to wait on and test so I prefer keeping this around.
For cherami backend itself, I want to iterate on logging vs error channels based on user feedback.